pennylane/circuit_graph.py

```diff
@@ -167,22 +167,29 @@ def __init__(self, ops, obs, wires, par_info=None, trainable_params=None):
 
         # TODO: State preparations demolish the incoming state entirely, and therefore should have no incoming edges.
 
-        self._graph = nx.DiGraph()  #: nx.DiGraph: DAG representation of the quantum circuit
+        self._graph = rx.PyDiGraph(
+            multigraph=False
+        )  #: rx.PyDiGraph: DAG representation of the quantum circuit
+
         # Iterate over each (populated) wire in the grid
         for wire in self._grid.values():
             # Add the first operator on the wire to the graph
             # This operator does not depend on any others
-            self._graph.add_node(wire[0])
+
+            if wire[0] not in self._graph.nodes():
+                self._graph.add_node(wire[0])
 
             for i in range(1, len(wire)):
                 # For subsequent operators on the wire:
-                if wire[i] not in self._graph:
+                if wire[i] not in self._graph.nodes():
                     # Add them to the graph if they are not already
                     # in the graph (multi-qubit operators might already have been placed)
                     self._graph.add_node(wire[i])
 
                 # Create an edge between this and the previous operator
-                self._graph.add_edge(wire[i - 1], wire[i])
+                self._graph.add_edge(
+                    self._graph.nodes().index(wire[i - 1]), self._graph.nodes().index(wire[i]), ""
```

Comment:
how come the `""` is needed?

Reply:
There isn't any default value for the edge-data in RX! You can check the doc [here](https://qiskit.org/documentation/retworkx/stubs/retworkx.PyDiGraph.add_edge.html#retworkx.PyDiGraph.add_edge).

Reply:
Ah, got it! I think this is another case that would benefit from a code comment

Reply:
Hi @josh146. Could you please check my comments before merging? :slightly_smiling_face: